### PR TITLE
Histogram Tool

### PR DESCRIPTION
**Pull request by bkallaher**
#43 is started with this PR. The script will use auto binning to determine the number of bins to generate for the plot. numpy version must be >=1.11.0 for the auto binning to work.

Example usage:
`python histogram.py /depth depth`
The ordering for the arguments is topic then what value from the message to use for the histogram.

### CODE REVIEW

**Review by skallaher — APPROVED** [review]
Looks good. Could this be expanded in future to plot multiple data sets on the same plot?

**Comment by bkallaher** [comment]
It doesn't seem that simple to get multiple sets on one plot due to the nature of getting data. You would have to dynamically generate callback functions for each subscriber to ensure that data isn't mixed up between sets. If that wasn't an issue the matplotlib hist will support multiple data sets in that manner.

**Comment by skallaher** [comment]
You could have a single node manager that spawns nodelets that then report back the raw data. This data would then be plotted by the manager. (These could all be separate nodes, but that would blow up pretty quickly if a lot of data is requested)

File: scripts/histogram.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    parser.add_argument('attribute', help="Attribute of the message to plot")
+    parser.add_argument('--normed', action='store_true', default=False,
+                        help="Normalize the data in the Histogram")
+    parser.add_argument('--color', default='blue',
+                        help="Choose the color of the bars in the plot")
+    parser.add_argument('--alpha', type=float, default=0.75,
+                        help="Alpha of the color to use")
+    parser.add_argument('--bins', default='auto',
+                        help="Number of bins to use for the histogram")
+    args = parser.parse_args()
```
Comment (irwineffect) [inline scripts/histogram.py:72]: Can we move the rospy.init_node() call to after parser.parse_args()?

**Review by irwineffect — APPROVED** [review]
This looks good! I would love if we could get it to update live once a second or so.

**Pull request merged**